pyproject.toml

```diff
@@ -36,36 +36,46 @@ classifiers = [
 ]
 readme = "README.md"
 dependencies = [
-  "click",
-  "cookiecutter",
-  "matplotlib",
-  "networkx",
   "numpy",
   "pandas",
-  "solara",
   "tqdm",
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
+# User collections
+rec = ["mesa[network,viz]"]
+all = ["mesa[network,viz,dev,examples,docs]"]
+# Core dependencies
+network = [
+  "networkx",
+]
+viz = [
+  "matplotlib",
+  "solara",
+]
+# Dev and CI stuff
 dev = [
-  "ruff~=0.1.1",   # Update periodically
-  "pytest >= 4.6",
+  "mesa[rec]",
+  "ruff",
+  "pytest",
   "pytest-cov",
   "sphinx",
   "pytest-mock",
+  "cookiecutter",
```

Comment:
Not sure whether this belongs in dev. Dev for me means further development of mesa itself. However, if I understand this correctly, it is used to setup a default project which uses mesa. So then it should be a core dependency.

Reply:
Yeah bit in doubt if we want to ship this by default. It has quite a lot of required dependencies:

> [arrow](https://pypistats.org/packages/arrow) | [binaryornot](https://pypistats.org/packages/binaryornot) | [click](https://pypistats.org/packages/click) | [jinja2](https://pypistats.org/packages/jinja2) | [python-slugify](https://pypistats.org/packages/python-slugify) | [pyyaml](https://pypistats.org/packages/pyyaml) | [requests](https://pypistats.org/packages/requests) | [rich](https://pypistats.org/packages/rich)

Of which some have their own dependencies, etc.

Related to https://github.com/projectmesa/mesa/issues/2331.

Reply:
Yes, cookiecutter and also click should be part of core mesa, since they provide the "mesa startproject / mesa run" command. While I think we should deprecate those in favor of GitHub templates for the scope of this PR they should belong to core

Reply:
I agree and would be in favor of just removing the cookie-cutter stuff for mesa 3.

Reply:
I'm going to investigate this before the next alpha.

Reply:
ok, then let's get this merged